src/test/regress/expected/merge.out

```diff
@@ -214,9 +216,45 @@ HINT:  To remove the local data, run: SELECT truncate_local_data_after_distribut
 
 (1 row)
 
+-- Updates one of the row with customer_id  = 30002
```

Comment:
Few more classes of queries you might consider adding to the test:
```SQL
drop table if exists source, target;
create table source (                                                                   
    id   integer, z int, d jsonb
);
create table target (
    id   integer, z int, d jsonb
);
insert into source SELECT i,i FROM generate_series(0,100)i;

select create_distributed_table('target','id'), create_distributed_table('source', 'id');

-- single shard query given source is filtered
-- and Postgres is smart to pushdown filter to the target as well
EXPLAIN MERGE INTO target sda USING (SELECT * FROM SOURCE WHERE id = 1) sdn ON
sda.id = sdn.id
WHEN NOT matched THEN
INSERT (id, z)
VALUES (sdn.id, 5);


-- zero shard query as filters do not match
EXPLAIN MERGE INTO target sda USING (SELECT * FROM SOURCE WHERE id = 1) sdn ON
sda.id = sdn.id AND sda.id = 2
WHEN NOT matched THEN
INSERT (id, z)
VALUES (sdn.id, 5);

-- zero shard query as source is zero shard
-- normally, I'd expect this to work with MERGE 0 tag, but anyway
EXPLAIN MERGE INTO target sda USING (SELECT * FROM SOURCE WHERE false) sdn ON
sda.id = sdn.id AND sda.id = 2
WHEN NOT matched THEN
INSERT (id, z)
VALUES (sdn.id, 5);
ERROR:  MERGE command is only supported when distributed tables are joined on their distribution column

-- join for source is happening at at different place
EXPLAIN MERGE INTO target sda USING source s1 LEFT JOIN (SELECT * FROM source) s2 USING(z)  ON
sda.id = s1.id  AND s1.id = s2.id
WHEN NOT matched THEN
INSERT (id, z)
VALUES (s2.id, 5);

-- update JSON column
EXPLAIN MERGE INTO target sda USING SOURCE sdn ON
sda.id = sdn.id
WHEN matched THEN
UPDATE SET d = '{"a" : 5}';


CREATE FUNCTION immutable_hash(int) RETURNS int
AS 'SELECT hashtext( ($1 + $1)::text);'
LANGUAGE SQL
IMMUTABLE
RETURNS NULL ON NULL INPUT;


MERGE INTO target sda USING SOURCE sdn ON
sda.id = sdn.id
WHEN matched THEN
UPDATE SET z = immutable_hash(sdn.z);

-- modifying CTE
EXPLAIN 
WITH cte_1 AS (DELETE FROM target)
MERGE INTO target sda USING source sdn ON
sda.id = sdn.id 
WHEN NOT matched THEN
INSERT (id, z)
VALUES (sdn.id, 5);
ERROR:  MERGE command is only supported when distributed tables are joined on their distribution column

```

Reply:
I've added them, please check to see if they match the expected output.